docs/design/908-proposal-process.md

```diff
@@ -0,0 +1,410 @@
+
+3. *Design document*: If requested, a design document is written, and the PR
+   for the design document becomes the new home for discussing the proposal.
+
+4. *Consensus*: Once comments and changes have slowed down, a final public
+   discussion aims to reach consensus for one of two outcomes:
+
+   * Accept the proposal, or
+   * decline the proposal.
+
+[new-proposal]: https://github.com/grafana/alloy/issues/new?assignees=&labels=proposal&projects=&template=proposal.yaml
+
+### Scope
+
+The proposal process is intended for user-facing features, enhancements or any
+largely scoped internal change. Other changes typically do not need a proposal,
+such as minor code refactors. If a contributor is uncertain about whether a
+proposal is needed, it is preferable to create a proposal instead of jumping
+into a change that may be declined.
+
+Proposals may initially be brief, with just enough detail to explain what is
+being proposed, why it is being proposed, and any relevant details that may
+guide the public discussion towards consensus.
+
+Proposal authors may skip straight to creating a design document for their
+proposal if they know one will be needed. If in doubt, it's better to create a
+proposal issue first.
+
+Any existing issue can be turned into a proposal by adding a `proposal` label.
+
+### Compatibility
+
+Proposals that break [backward compatibility] are moved to a "Hold" state until
+work on the next major release begins. As major releases are infrequent, this
+avoids approved proposals from becoming obsolete before they are implemented.
+
+[backward compatibility]: https://grafana.com/docs/alloy/latest/introduction/backward-compatibility/
+
+### Design documents
+
+Proposals that require more careful consideration and explanation require a
+design document.
+
+* Design documents should be checked into the Alloy repository as
+  `docs/design/NNNN-name.md`, where `NNNN` is the GitHub issue number of the
+  proposal and `name` is a short descriptive name deliminated by hyphens.
+
+* Design documents should follow the template in `docs/design/template.md`.
+
+* Design documents should address any specific concerns raised during the
+  initial discussion.
+
+* Design documents should acknowledge and address alternative solutions, if
+  known, to the problem statement.
+
+Once a design document is created, the proposal author should:
+
+* Close the original proposal issue and link to the design document PR.
+* Update the proposal issue description with a link to the design document.
+
+Closing the original proposal issue in favor of the design document PR prevents
+fragmenting discussions as contributors work towards consensus on a proposal.
+
+Design documents do not track the state of the proposal; the proposal PR  is
+the source of truth.
+
+The PR for design documents are merged once the associated proposal becomes
+_Accepted_. If the proposal becomes _Declined_, the PR for the design document
```

Comment:
My concern about not closing the issue is that the proposal review team will have to continue to monitor both the issue and the design document PR to make sure discussions are happening in the right place, generating more work. 

Do you have any other ideas of how we can ensure discussions happen in one place at a time?

Reply:
Yeah, we can lock conversation on the proposal issue, so the comments won't happen. But we can still use labels etc to track status using the issue ticket maybe? And the PR being just a secondary thing.

![image](https://github.com/grafana/alloy/assets/17101802/a4d1249d-3ec8-4cf0-9b3a-fddf3c7b30b2)

Reply:
I'm not sure about this; with closing an issue, the author of the proposal can close it immediately themselves, but only maintainers can lock an issue once they notice a proposal has moved to a design document. I wonder if this might cause too much extra work for maintainers, and risks someone forgetting to lock an issue. 

What do you think about trying the process in the current state (where discussion moves to a PR by closing the original issue, if a design document is requested), seeing how it works in practice, and then making changes if necessary?

Reply:
Sounds good!